zebedee-cms/src/main/java/com/github/onsdigital/zebedee/service/ImageServiceImpl.java

```diff
@@ -33,40 +35,64 @@ public ImageServiceImpl(ImageClient imageClient) {
      * Publish the images contained in the given collection.
      */
     @Override
-    public void publishImagesInCollection(Collection collection) throws IOException, ImageAPIException {
+    public ImageServicePublishingResult publishImagesInCollection(Collection collection) throws IOException, ImageAPIException {
 
         Images images = imageClient.getImages(collection.getId());
         info().data("publishing", true).data("collectionId", collection.getId())
                 .data("numberOfImages", images.getTotalCount())
                 .log("publishing images in collection");
 
+        ImageServicePublishingResult result = new ImageServicePublishingResult(images.getTotalCount());
         for (Image image : images.getItems()) {
-            publishImage(image, collection.getId());
+            PublishStatus status = publishImage(image, collection.getId());
+            if (PublishStatus.UNPUBLISHED.equals(status)) {
+                result.addUnpublishedImage(image.getId(), image.getState());
```

Comment:
Is there any kind of "reason" type data that could be included in this? Just wondering if its available and if it would be useful. Possibly out of scope but asking the question.

Reply:
I was keeping it as small as possible. This is purely for use in reporting to slack so as little info as possible to keep the message small. We could always expand another time if it's deemed necessary.